### PR TITLE
WSTEAMA-220 - Refactor the structure of the message banner component to only accept heading, description, link and image

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves #WSTEAMA-220

**Overall change:**
Refactoring the message banner component, so that it does not render multiple promos, and ensuring that the implementation is not tied to the Curation data returned from TIPO.

**Code changes:**
Update the message banner, associated tests, storybook entries and references

**Useful Links**
[View Test Page on Storybook](https://wsteama-220-refactor-message-banner-structure--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=topic-page--kyrgyz-with-message-banners&viewMode=story)
[Kyrgyz Test Page (localhost)](http://localhost:7080/kyrgyz/topics/cvpv9djp9qqt?renderer_env=test) 

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/components/MessageBanner/index.test.tsx [hunk 1]
```diff
@@ -1,97 +1,119 @@
...
-      <MessageBanner
-        summaries={summaries}
-        title={kyrgyzBanner.title}
-        position={kyrgyzBanner.position}
-      />,
-    );
-    const region = screen.getByRole('region');
-    expect(region).toBeInTheDocument();
-  });
+  describe('One summary', () => {
```
Comment (karinathomasbbc) [inline src/app/components/MessageBanner/index.test.tsx]:
```suggestion
  describe('which has a curation with one summary', () => {
```

File: src/app/components/MessageBanner/index.test.tsx [hunk 2]
```diff
@@ -1,97 +1,119 @@
...
+          summary={summary}
+          title={kyrgyzMessageBannerOnePromo.title}
+          position={kyrgyzMessageBannerOnePromo.position}
+        />,
+      );
+      const image = screen.getByAltText('');
+      expect(image).toBeInTheDocument();
+    });
   });
+  describe('Two summaries', () => {
```
Comment (karinathomasbbc) [inline src/app/components/MessageBanner/index.test.tsx]:
```suggestion
  describe('which has a curation with two summaries', () => {
```

File: src/app/components/MessageBanner/index.test.tsx [hunk 3]
```diff
@@ -1,97 +1,119 @@
...
+    const { summaries } = kyrgyzMessageBannerTwoPromos;
+    it('should only render one banner per curation', () => {
```
Comment (karinathomasbbc) [inline src/app/components/MessageBanner/index.test.tsx]:
```suggestion
    it('should only render the first summary', () => {
```

Comment (karinathomasbbc) [reply src/app/components/MessageBanner/index.test.tsx]: Hmmm wondering if this test still makes sense here - because we are passing in `summaries[0]` and because typescript will prevent us from passing in invalid data... 

I think we should consider moving this test to the Topic page (and tweaking it), and checking that only the first summary in the data is rendered, and not the second one.

Comment (pvaliani) [reply src/app/components/MessageBanner/index.test.tsx]: I know it's ironic but

...Karina pls sleep 🤣

Comment (karinathomasbbc) [reply src/app/components/MessageBanner/index.test.tsx]: So thinking more about this test - we have a very similar ones in https://github.com/bbc/simorgh/blob/latest/src/app/pages/TopicPage/index.test.jsx#L167-L185 or https://github.com/bbc/simorgh/blob/latest/src/app/pages/TopicPage/index.test.jsx#L187-L198, and I am wondering if we could do something like this:

```
/**
 ** Find curation from the kyrgyzTopicWithMessageBanners with visual style = BANNER 
 ** and visual prominence = NORMAL AND summaries array length greater than 1   
 ** 
 ** Query ALL by role based on the curation title (getByRole only gets the first one if it exists)
 ** Ensure the length of the array is 1
 **/
```

File: src/app/components/MessageBanner/index.test.tsx [hunk 4]
```diff
@@ -1,97 +1,119 @@
...
-        title={kyrgyzBanner.title}
-        position={kyrgyzBanner.position}
-      />,
-    );
-    const region = screen.getByRole('region');
-    expect(region).toBeInTheDocument();
-  });
+  describe('One summary', () => {
+    const { summaries } = kyrgyzMessageBannerOnePromo;
+    const summary = summaries[0];
```
Comment (karinathomasbbc) [inline src/app/components/MessageBanner/index.test.tsx]: If we wanted to, we could simplify this into a single line (unless linter complains or suggests something else)
```suggestion
    const summary = kyrgyzMessageBannerOnePromo.summaries[0];
```

File: src/app/components/MessageBanner/index.tsx [hunk 5]
```diff
@@ -12,12 +12,12 @@ import styles from './index.styles';
...
 
 interface MessageBannerProps {
-  summaries: Summary[];
+  summary: Summary;
   title: string;
   position: number;
 }
 
-const MessageBanner = ({ summaries, title, position }: MessageBannerProps) => {
+const MessageBanner = ({ summary, title, position }: MessageBannerProps) => {
```
Comment (karinathomasbbc) [inline src/app/components/MessageBanner/index.tsx]: Is this also an opportunity to rename `title` to something like `heading` to prevent confusion with the summary title?
```suggestion
const MessageBanner = ({ summary, heading, position }: MessageBannerProps) => {
```

Comment (karinathomasbbc) [inline src/app/components/MessageBanner/index.tsx]: We also need to update the docs! https://github.com/bbc/simorgh/blob/latest/src/app/components/MessageBanner/README.md#messagebanner

File: src/app/components/MessageBanner/index.test.tsx (hunk 3 above)
Comment (karinathomasbbc) [reply src/app/components/MessageBanner/index.test.tsx]: Thanks for sorting this @louisearchibald 🧪

File: src/app/components/MessageBanner/index.tsx [hunk 6]
```diff
@@ -12,64 +10,59 @@ import styles from './index.styles';
...
 
 interface MessageBannerProps {
-  summaries: Summary[];
-  title: string;
-  position: number;
+  heading: string;
+  description: string;
+  link: string;
+  linkText: string;
+  image: string;
```
Comment (amoore108) [inline src/app/components/MessageBanner/index.tsx:17]: May be worth making this optional as its conditionally rendered. Could also change the readme to indicate its optional and maybe have a test case that the image _shouldn't_ be shown if its undefined/null.

**Comment by amoore108** [comment]
Nice refactor, great work by all involved 👍🏼 Just a comment around the image being optional, but all good!

File: src/app/components/MessageBanner/index.tsx (hunk 6 above)
Comment (karinathomasbbc) [reply src/app/components/MessageBanner/index.tsx:17]: 👋 @amoore108 thanks for the review! So the designs for the message banner allowed for description and image to be optional, however the data from Tipo means that the description (which is really the promo description) and image url (promo image) will always be there. It is not possible to overtype the description / image url with an empty string in Tipo.

For that reason, we decided to leave description and image url as mandatory fields. If in the future there is another use case for the message banner, where image/description are not required, then perhaps we could do this then (and add the appropriate tests?)

Comment (amoore108) [reply src/app/components/MessageBanner/index.tsx:17]: Ah cool, no worries that makes sense! Thanks for clarifying 😄

**Review by amoore108 — APPROVED** [review]

File: src/app/components/MessageBanner/README.md [hunk 7]
```diff
@@ -2,35 +2,28 @@
...
-```
-
-You can add a ternary condition around the `MessageBanner` to select curations with a `visualStyle` of `BANNER`
-
 ```tsx
 {
-  visualStyle === 'BANNER' && (
-    <MessageBanner title={curationTitle} summaries={summaries} />
-  );
+  <MessageBanner
```
Comment (pvaliani) [inline src/app/components/MessageBanner/README.md:21]: We'll need to update this once the `TopicPage` refactor is done

File: src/app/components/MessageBanner/index.tsx (hunk 6 above)
Comment (pvaliani) [inline src/app/components/MessageBanner/index.tsx:17]: Does `image` need to be optional? i.e `image?`. I can't recall whether it was finally decided that there may be no image on a banner

File: src/app/pages/TopicPage/index.test.jsx [hunk 8]
```diff
@@ -222,5 +222,23 @@ describe('Topic Page', () => {
         0,
       );
     });
+
+    it('should only render the first summary if there are more than one', () => {
```
Comment (pvaliani) [inline src/app/pages/TopicPage/index.test.jsx]: Nitpicky but maybe rephrase this to `should only render the first summary if there is more than one summary in the curation`?

**Review by pvaliani — COMMENTED** [review]
Nice work!

File: src/app/components/MessageBanner/index.tsx (hunk 6 above)
Comment (karinathomasbbc) [reply src/app/components/MessageBanner/index.tsx:17]: Even though the designs had the image as optional, we realised that the data always had an image. Aaron had a similar comment, which I replied to already - https://github.com/bbc/simorgh/pull/10581#discussion_r1099840177 - but the TLDR was, let's leave it as required for now, and next time the message banner is required on a non-Topic page, then we can think about making it optional, and adding the appropriate tests then.

Comment (pvaliani) [reply src/app/components/MessageBanner/index.tsx:17]: Sweet just had a read,  nice one

**Review by pvaliani — APPROVED** [review]
